package com.back.shard.cash.dto;

import com.back.boundedContext.cash.domain.CashMember;
import com.back.boundedContext.member.domain.Member;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.time.LocalDateTime;

@Getter
@AllArgsConstructor
public class CashMemberDto {
    private final Integer id;
    private final String username;
    private final String nickname;
    private final int activityScore;
    private final LocalDateTime createDate;
    private final LocalDateTime modifyDate;

    public CashMemberDto(CashMember member) {
        this(
                member.getId(),
                member.getUsername(),
                member.getNickname(),
                member.getActivityScore(),
                member.getCreateDate(),
                member.getModifyDate()
        );
    }
}
